addons/mail/static/src/messaging/component/composer_text_input/composer_text_input.js

```diff
@@ -102,13 +161,35 @@ class ComposerTextInput extends Component {
      * Returns selection start position.
      *
      * @private
-     * @returns {integer}
+     * @returns {number}
      *
      */
     _getSelectionStart() {
         return this._textareaRef.el.selectionStart;
     }
 
+    /**
+     * @private
+     * @param selection
+     */
+    _insertMentionSuggestion(selection) {
+        const cursorPosition = this._getSelectionStart();
+        const textLeft = this.getContent().substring(0, cursorPosition - (this._mentionWord.length));
+        const textRight = this.getContent().substring(cursorPosition, this.getContent().length);
+        const insert = selection.name.replace(/ /g, '\u00a0');
+        this._textareaRef.el.value = textLeft + insert + ' ' + textRight;
+        this._textareaRef.el.setSelectionRange(
+            textLeft.length + selection.name.length + 2,
+            textLeft.length + selection.name.length + 2
+        );
+        const mention = this._mentions.find(item => item.id === selection.id);
+        if (!mention) {
+            this._mentions.push(selection);
+        }
+        this.saveStateInStore();
```

Comment:
I feel like insertion should be in store (i.e. in `mail.messaging.entity.Composer`)